hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/BlockMetaSourceProvider.java

```diff
@@ -0,0 +1,32 @@
+package com.hedera.mirror.web3.evm.properties;
+
+import java.time.Instant;
+import javax.inject.Singleton;
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.evm.frame.BlockValues;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.hedera.mirror.web3.exception.InvalidParametersException;
+import com.hedera.mirror.web3.repository.RecordFileRepository;
+
+@Singleton
+public class BlockMetaSourceProvider {
+
+    @Autowired
+    private RecordFileRepository recordFileRepository;
+
+    public BlockMetaSourceProvider(final RecordFileRepository recordFileRepository) {
+        this.recordFileRepository = recordFileRepository;
+    }
```

Comment:
Use constructor injection and Lombok
```suggestion
@Singleton
@RequiredArgsConstructor
public class BlockMetaSourceProvider {

    private final RecordFileRepository recordFileRepository;
```